### PR TITLE
python310Packages.einops: disable failing tests

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes

See https://github.com/NixOS/nixpkgs/issues/174872

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/einops/default.nix [hunk 1]
```diff
@@ -26,38 +31,47 @@ buildPythonPackage rec {
...
-    export HOME=$TMPDIR
-
+  preCheck = ''
+     export HOME=$(mktemp -d);
+    
     # Prevent hangs on PyTorch-related tests, see
     # https://discuss.pytorch.org/t/pytorch-cpu-hangs-on-nn-linear/17748/4
-    export OMP_NUM_THREADS=1
-
-    nosetests -v -w tests
```
Comment (yl3dy) [inline pkgs/development/python-modules/einops/default.nix:54]: Why removing `nosetests` in favor of `pytest`? The upstream recommends `nose` (also it is still in the argument list).

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 174873` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.augmax</li>
    <li>python39Packages.augmax</li>
    <li>python39Packages.elegy</li>
    <li>python39Packages.treex</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.einops</li>
    <li>python39Packages.arviz</li>
    <li>python39Packages.einops</li>
    <li>python39Packages.xarray-einstats</li>
  </ul>
</details>

File: pkgs/development/python-modules/einops/default.nix [hunk 2]
```diff
@@ -26,38 +31,47 @@ buildPythonPackage rec {
...
 
-  checkPhase = ''
-    export HOME=$TMPDIR
-
+  preCheck = ''
+     export HOME=$(mktemp -d);
+    
     # Prevent hangs on PyTorch-related tests, see
     # https://discuss.pytorch.org/t/pytorch-cpu-hangs-on-nn-linear/17748/4
-    export OMP_NUM_THREADS=1
```
[the remark this answers is not in the document]
Comment (fabaff) [reply pkgs/development/python-modules/einops/default.nix:52]: Thanks for the feedback.

File: pkgs/development/python-modules/einops/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/einops/default.nix:54]: `nose` is unmaintained for 7 years now and the "pytest eco-system" in nixpkgs has features which makes the handling of tests simpler.

Comment (yl3dy) [reply pkgs/development/python-modules/einops/default.nix:54]: Seems fair then.

**Comment by fabaff** [comment]
@ofborg build python39Packages.elegy python39Packages.augmax

**Pull request merged**